### PR TITLE
Hdc1008 sensor driver

### PR DESCRIPTION
**Pull request by iceaway**
## Summary
New driver for the HDC1008 temperature/humidity sensor. Not sure about the copyright notice in the file header after the apache move, please let me know if I should change it.

## Testing
Tested on a stm32f103 (bluepill) board. Example application in a separate pull request to the apps repo.

### CODE REVIEW

File: include/nuttx/sensors/hdc1008.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline include/nuttx/sensors/hdc1008.h:33]: Sorry, but as an Apache project, all new files must have the Apache 2.0 header.  Could you please copy/paste the header from, sched/sched/sched_getcpu.c?  No copyrights or authors in the the Apache header.

File: drivers/sensors/hdc1008.c [hunk 2]
```diff
@@ -0,0 +1,1024 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline drivers/sensors/hdc1008.c:35]: Sorry, but as an Apache project, all new files must have the Apache 2.0 header.  Could you please copy/paste the header from, sched/sched/sched_getcpu.c?  No copyrights or authors in the the Apache header.

**Comment by iceaway** [comment]
Both files have been updated with the apache 2.0 header, please check again if it looks ok.

**Comment by patacongo** [comment]
> 
> 
> Both files have been updated with the apache 2.0 header, please check again if it looks ok.

Thank!  I will merge when the checks compelte.

**Comment by patacongo** [comment]
I don't think we need to wait for all of the PR checks.  They all passed last time.

**Pull request merged**